p2p/src/net/libp2p/discovery/mdns.rs

```diff
@@ -38,9 +38,15 @@ pub enum Mdns {
 }
 
 impl Mdns {
-    pub async fn new(enabled: bool) -> Self {
+    pub async fn new(enabled: bool, query_interval: u64) -> Self {
```

Comment:
It's probably better to pass the argument here as a Duration, so that no ambiguity about units happen. I understand that the argument is a number, but the argument has to identify the unit, and then there should be a proper conversion there to this function as a Duration. That's probably the cleanest solution to time issues.

Reply:
`std::time::Duration` will convey as little information about the units used when reading the code but now the conversion is not happening here but at the call site (or probably when the argument is parsed).

Reply:
I think I don't understand your proposal for fixing this, could you explain it in more detail? Just changing the parameter type doesn't fix the issue of the unit being ambiguous as I explained in my previous comment so I'm not entirely sure how you want this information displayed.